cli/cmd/uninstall.go

```diff
@@ -25,18 +28,34 @@ func newCmdUninstall() *cobra.Command {
 This command provides all Kubernetes namespace-scoped and cluster-scoped resources (e.g services, deployments, RBACs, etc.) necessary to uninstall Linkerd control plane.`,
 		Example: ` linkerd uninstall | kubectl delete -f -`,
 		RunE: func(cmd *cobra.Command, args []string) error {
-			return uninstallRunE(cmd.Context())
+
+			k8sAPI, err := k8s.NewAPI(kubeconfigPath, kubeContext, impersonate, impersonateGroup, 0)
+			if err != nil {
+				return err
+			}
+
+			if !force {
+				podsList, err := k8sAPI.CoreV1().Pods("").List(cmd.Context(), metav1.ListOptions{})
+				if err != nil {
+					return err
+				}
+				for _, pod := range podsList.Items {
+					// Skip core control-plane and extension namespaces when checking for proxies
+					if !strings.HasPrefix(pod.Namespace, "linkerd") && healthcheck.ContainsProxy(pod) {
```

Comment:
I think you can tap here into the `linkerd-namespace` flag instead of relying on the hard-coded value. Also, according to the requirement in #5622, this should also check for injected pods in extensions, so a strict comparison against the linkerd namespace might be enough.